File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        result = 0
        left, right = 0, len(heights) - 1
        #heights = sorted(heights)
        print(heights)

        max_area = (right - left) * min(heights[left], heights[right])
        while left < right:
            if heights[left] < heights[right]:
                left += 1
                current_max = (right - left) * min(heights[left], heights[right])
                if current_max > max_area:
                    max_area = current_max
                print(max_area)
            elif heights[left] >= heights[right]:
                right -= 1
                current_max = (right - left) * min(heights[left], heights[right])
                if current_max > max_area:
                    max_area = current_max
                print(max_area)
            #else:
                #print(current_max)
                #break
        
        return max_area        
                                 
        

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    private int count = 0, result = 0;
    private boolean found = false;

    public int kthSmallest(TreeNode root, int k) {

        this.dfs(root, k);
        return this.result;
    }

    public void dfs(TreeNode node, int k) {

        if(node == null || found) {return;}

        this.dfs(node.left, k);

        this.count++;
        if(this.count == k) {
            this.result = node.val;
            this.found = true;
            return;
        }

        this.dfs(node.right, k);
    }
}
